### PR TITLE
Place startup.log and status in /var/lib/icinga2/api, not /var/lib/icinga2/api/zones-stage

### PR DESCRIPTION
**Pull request by Al2Klimov**
... not to loose them.

### CODE REVIEW

**Comment by Al2Klimov** [comment]
Shall we include this in v2.12.0?

**Comment by julianbrost** [comment]
What's the situation where you'd lose these files? Shouldn't this only happen when the validation was successful and Icinga reloads (which should log roughly the same information, but for the actually running instance)?

**Comment by Al2Klimov** [comment]
I’m not sure. Once a user was about to share the files, but [they vanished](https://github.com/Icinga/icinga2/issues/7742#issuecomment-605933338).

**Comment by Al2Klimov** [comment]
@cla-bot check

**Comment by julianbrost** [comment]
I think `zones-stage/startup.log` should only be missing when a new incoming config update was received and
https://github.com/Icinga/icinga2/blob/44a34cf070d3c76c198a15eaf3ecd5d6acf5b7cc/lib/remote/apilistener-filesync.cpp#L328
clears `zones-stage/` quite early:
https://github.com/Icinga/icinga2/blob/44a34cf070d3c76c198a15eaf3ecd5d6acf5b7cc/lib/remote/apilistener-filesync.cpp#L359-L366

But if that happens another should happen unless the config is identical to the running config. So I think if the current contents of `zone-stage/` are broken, either a config update should still be written to this directory or a validation for the contents should be running, so there should be a new `startup.log` soon.

Then there's a second part to this like in https://github.com/Icinga/icinga2/issues/7742#issuecomment-605933338: If there's an error about this somewhere in the logs, the referenced file might not exist. This only gets somewhat better with this PR: the file should always exist, but it may contain the log from a later successful validation.

So would this PR be an improvement? I think slightly because it makes the log a little less confusing: the referenced file will exist and you can see from the timestamps in the file that is was overwritten with something newer than from what was in there when the log message was generated. But I don't think it will help retrieving a log that you actually care for. If I'm not missing something, in all situations that resulted in a missing file before, now it would be overwritten with a successful later validation (unless there's an incoming config update going on right at that moment).

The alternative would be to write files with timestamps in their name, similar to how it's done for crash logs. But that would require extra thoughts about whether those files should be cleaned up automatically and so on. But if the goal is to actually retrieve logs from previous failed validations, I think something like that would be required.

So the question is: what's the goal here? Is making the logs a little less confusing good enough? I think this PR was started because there was an interest in logs from a previous failed validation and I fear this won't probably help for this.

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: lib/remote/apilistener-filesync.cpp [hunk 1]
```diff
@@ -592,15 +592,16 @@ void ApiListener::TryActivateZonesStage(const std::vector<String>& relativePaths
...
 
+	String apiDir = GetApiDir();
 	String apiZonesDir = GetApiZonesDir();
 	String apiZonesStageDir = GetApiZonesStageDir();
 
-	String logFile = apiZonesStageDir + "/startup.log";
+	String logFile = apiDir + "/zones-stage-startup.log";
 	std::ofstream fpLog(logFile.CStr(), std::ofstream::out | std::ostream::binary | std::ostream::trunc);
 	fpLog << pr.Output;
 	fpLog.close();
```
[the remark this answers is not in the document]
Comment (Al2Klimov) [reply lib/remote/apilistener-filesync.cpp:602]: I have a better idea...

**Review by julianbrost — APPROVED** [review]

**Pull request merged**